mypy/checker.py

```diff
@@ -984,7 +984,12 @@ def is_unannotated_any(t: Type) -> bool:
         check_incomplete_defs = self.options.disallow_incomplete_defs and has_explicit_annotation
         if show_untyped and (self.options.disallow_untyped_defs or check_incomplete_defs):
             if fdef.type is None and self.options.disallow_untyped_defs:
-                self.fail(message_registry.FUNCTION_TYPE_EXPECTED, fdef)
+                if (not len(fdef.arguments) or
+                        (len(fdef.arg_names) == 1 and 'self' in fdef.arg_names)):
```

Comment:
@JukkaL or @ilevkivskyi Is there a more reliable way to tell whether something's a method? Some (unusual) code bases use a different name instead of `self`, and class methods also aren't covered by this test.

It's also possible to scan the body to see whether occurrences of `return` have a value or not on them (though I don't know how practical that is from this context).

Reply:
I think for `self` there is actually one: `Var.is_self`. I am not sure we have the same for `cls`.

Reply:
Thanks! I tried checking for `fdef.arguments[0].variable.is_self` here but it seems like I always get `False`. I couldn't find where `is_self` was being set-- for now I have this checking for the strings `self` and `cls` but agree it's not great

Reply:
It may be because if a function is missing type annotation, then we don't analyze it at all during semantic analysis. I think it is OK to rely on the argument name, since this is needed only to tweak the error message, and _not_ to decide whether we need to emit the error.